### DISCUSSION TITLE
Possible transpillation issue 83/101 versions

### DISCUSSION THREAD

**Discussion by dantaeusb** [swc-project/swc#8541] [Q&A]
Building poppperjs library (particularly this code https://github.com/floating-ui/floating-ui/blob/master/packages/dom/src/platform/getClippingRect.ts) with SWC 1.3.83 (left) and 1.3.101 (right)

![Screenshot 2024-01-23 at 04 27 15](https://github.com/swc-project/swc/assets/4356112/27a0ac21-0f1d-44a5-a75f-92caaff77e16)

On the left, `a` is first initialized, then reduced.
On the right, `a` is initialized inside the accumulator of `a.reduce`, which causes "Cannot read properties of undefined (reading 'reduce')"

I'm not entirely sure that it's caused by the SWC, but the code does change if I rollback to 83. Cannot yet create a minimal reproducible repo but seems serious enough to raise this question. 

I'm going to catch some sleep to make sure I'm in a sane state, but looks like the code on the left cannot ever be run successfully, as `a` is not defined, and while `reduce` is called before `a` initialized in argument, it cannot work that way?

**Comment by kdy1** [comment]
I think it's a regression and it's a bug. Are you using swc minifier?

**Reply by dantaeusb** [reply]
Yep, with the following options:
```js
  optimization: {
    minimize: true,
    minimizer: [
      new TerserPlugin({
        minify: TerserPlugin.swcMinify,
        terserOptions: {
          mangle: true,
        },
      }),
    ],
  },
```

**Reply by kdy1** [reply]
Can you file an issue?
